dataactbroker/handlers/fileHandler.py

```diff
@@ -703,31 +694,14 @@ def add_job_info_for_d_file(self, upload_file_name, timestamped_name, submission
             job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
             val_job.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
             val_job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
-            # Generate random uuid and store generation task
-            task_key = uuid4()
-            task = FileGenerationTask(generation_task_key=task_key, job_id=job.job_id)
-            sess.add(task)
-            sess.commit()
         except ValueError as e:
             # Date was not in expected format
             exc = ResponseException(str(e),StatusCode.CLIENT_ERROR,ValueError)
             return False, JsonResponse.error(exc, exc.status, url = "", start = "", end = "",  file_type = file_type)
 
-        if not self.isLocal:
-            # Create file D API URL with dates and callback URL
-            get_url = FileHandler.get_d_file_url(task_key, file_type_name, cgac_code, start_date, end_date)
+        result = self.call_d_file_api(file_type_name, cgac_code, start_date, end_date, job)
 
-            _debug_logger.debug('Calling D file API => %s', get_url)
-            try:
-                if not self.call_d_file_api(get_url):
-                    self.handleEmptyResponse(job, val_job)
-            except Timeout as e:
-                exc = ResponseException(str(e), StatusCode.CLIENT_ERROR, Timeout)
-                return False, JsonResponse.error(e, exc.status, url="", start="", end="", file_type=file_type)
-        else:
-            self.complete_generation(task.generation_task_key, file_type)
-
-        return True, None
+        return True, None if result is None else result
```

Comment:
I've updated the return value to only return an exception (if running into a Timeout exception) and then this statement will work and push the error upwards as it did before. It will look cleaner as well.